tracer/test/Datadog.Trace.Debugger.IntegrationTests/DebuggerManagerDynamicTests.cs

```diff
@@ -0,0 +1,301 @@
+// <copyright file="DebuggerManagerDynamicTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Debugger.IntegrationTests.Assertions;
+using Datadog.Trace.Debugger.IntegrationTests.Helpers;
+using Datadog.Trace.Debugger.Sink;
+using Datadog.Trace.TestHelpers;
+using Samples.Probes.TestRuns.SmokeTests;
+using VerifyXunit;
+using Xunit;
+using Xunit.Abstractions;
+
+#nullable enable
+namespace Datadog.Trace.Debugger.IntegrationTests;
+
+#if !NETCOREAPP2_1
+[CollectionDefinition(nameof(DebuggerManagerDynamicTests), DisableParallelization = true)]
+[Collection(nameof(DebuggerManagerDynamicTests))]
+[UsesVerify]
+public class DebuggerManagerDynamicTests : TestHelper
+{
+    private const string LogFileNamePrefix = "dotnet-tracer-managed-";
+
+    // Log messages to verify dynamic state changes
+    private const string DynamicInstrumentationEnabledLogEntry = "Initializing Dynamic Instrumentation";
+    private const string ExceptionReplayEnabledLogEntry = "Initializing Exception Replay";
+    private const string CodeOriginForSpansEnabledLogEntry = "Initializing Code Origin for Spans";
+    private const string ApplyingDynamicDebuggerConfigLogEntry = "Applying new dynamic debugger configuration";
+    private const string DisabledByRemoteConfiguration = "is disabled by remote enablement.";
+    private const string TracerInitialized = "The profiler has been initialized";
+    private const string DebuggerConfigurationInitialized = "DATADOG DEBUGGER CONFIGURATION";
+
+    public DebuggerManagerDynamicTests(ITestOutputHelper output)
+        : base("Probes", Path.Combine("test", "test-applications", "debugger"), output)
+    {
+        SetServiceVersion("1.0.0");
+    }
+
+    [SkippableFact]
+    [Trait("Category", "EndToEnd")]
+    [Trait("Category", "ArmUnsupported")]
+    [Trait("RunOnWindows", "True")]
+    [Trait("Category", "LinuxUnsupported")]
+    public async Task DebuggerManager_DynamicInstrumentation_StartDisabled_EnabledViaRemoteConfig()
+    {
+#if NET8_0_OR_GREATER
+        // These tests often hang on x86 on .NET 8+. Needs investigation
+        Skip.If(!EnvironmentTools.IsTestTarget64BitProcess());
+#endif
```

Comment:
Shouldn't we investigate that _now_, before we merge 😟 I feel a bit uncomfortable knowing we have unknown hangs on all the MS _supported_ versions of .NET 😬

Reply:
Trying to run without skipping. I'll keep you posted.
b35dfcc3872e2aebc1c9c18d2e5b8b12ba381e83